### PR TITLE
OpenEXR: Add 3.3.2

### PR DESCRIPTION
**Pull request by irieger**
### Summary
Changes to recipe:  **openexr/3.3.2**

#### Motivation
Contains small fixes and performance improvements.

#### Details
https://github.com/AcademySoftwareFoundation/openexr/releases/tag/v3.3.2

Just needed adaptation of the data source, no patching etc.

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/openexr/3.x/conandata.yml
Comment (jcar87) [inline recipes/openexr/3.x/conandata.yml:3]: Please prefer to use `https://github.com/AcademySoftwareFoundation/openexr/releases/download/v3.3.2/openexr-3.3.2.tar.gz`  as that is guaranteed to be hash-stable (see https://github.com/orgs/community/discussions/46034)

Also note that since this release was published about ~2 hours ago - we will wait 24-48 hours before merging - at the scale of Conan Center - we see that during the first day or two that's when it is most likely that a maintainer will re-tag/re-call/re-publish a version, invalidating the checksum (unless you are the maintainer and can advise better).

Comment (jcar87) [reply recipes/openexr/3.x/conandata.yml]: Thanks! :)

**Review by AbrilRBS — APPROVED** [review]
Thanks!

**Review by jcar87 — APPROVED** [review]

**Pull request merged**
